doc/topics/cloud/libvirt.rst

```diff
@@ -106,6 +106,21 @@ When using ``ip-learning`` the IP is requested from libvirt. This needs a recent
 version and may only work for NAT networks. Another option is to use ``qemu-agent`` this requires
 that the qemu-agent is installed and configured to run at startup in the base domain.
 
+SSH Key Authentication
+======================
+Instead of specifying a password, an authorized key can be used for the minion setup. Ensure that
+the root user of your base image has the public key you want to use in .ssh/authorized_keys, then
```

Comment:
a more complex example using sudo and a non root user might be more helpful similar to the config given here: https://github.com/saltstack/salt/issues/40965#issue-225232406
```yaml
#provider
digitalocean-lampros:
  driver: digital_ocean
  ssh_key_name: digital-ocean-salt-cloud.pub
  ssh_key_file: /root/.ssh/id_rsa 
  ssh_username: tools
  ssh_interface: private
  ssh_port: 4910
  sudo: True 
  sudo_password: "--redacted--"
  personal_access_token: --redacted--
  location: San Francisco 2
  minion:
    master: salt 
```
an example with root as well would be good.  Ideally we should have examples that show all options that could be configured.

Reply:
Sorry for long silence! This PR was for libvirt specifically - I had trouble with other auth methods here - the others work fine like your example, libvirt not so much

Reply:
What shall we do here guys - the doc is "Getting started with Libvirt", I dont think any other provider makes sense here

Reply:
If @Akm0d has time to verify these, that would be ideal.  I probably take a look some time this month if not.

Reply:
I didn't mean other providers, I just meant document use of alternative ports, usernames, sudo usage, etc.  Any option that can be used should be documented.

Reply:
ok, i put this in because i wanted to use key rather than password login and figured others might, but we can do all libvirt providers options as well - where do i look up every single option so it can be documented?

Reply:
You would likely have to look through the code to get them all, you could get most of them by looking at the documentation for other providers and seeing what works.

Reply:
If you got most of them, that would probably be enough to merge for now.  Mainly it's just misleading in your pr as it is now to make people think that logging in as root is the only way possible.